driver/connect.c

```diff
@@ -1020,6 +1023,25 @@ SQLRETURN config_dbc(esodbc_dbc_st *dbc, esodbc_dsn_attrs_st *attrs)
 
 	if (attrs->cloud_id.cnt && (! decode_cloud_id(dbc, attrs))) {
 		goto err;
+	} else if (http_prefix.cnt <= attrs->server.cnt) {
+		/* make sure that user didn't input a HTTP URL (common mistake):
+		 * the error libcurl returns is not particularly descriptive */
+		do {
+			prefix = attrs->server;
+			prefix.cnt = http_prefix.cnt;
+			if (! EQ_CASE_WSTR(&prefix, &http_prefix)) {
+				prefix.cnt = https_prefix.cnt;
```

Comment:
Is it a problem if `https_prefix.cnt` > `attrs->server.cnt`?

Reply:
My initial thought was that "no", it's safe, since each attribute has a large enough buffer (currently 1K). But if somebody names their (locally resolvable) server `https` and the following characters in the buffer happen to be `://`, the error case would wrongly be triggered. I've fixed it now, thanks.